providers/oidc.go

```diff
@@ -123,6 +133,31 @@ func (p *OIDCProvider) enrichFromProfileURL(ctx context.Context, s *sessions.Ses
 	return nil
 }
 
+// enrichFromIntrospectURL enriches a session's claims and permissions via the JSON response of
+// an OIDC Introspection URL
+func (p *OIDCProvider) enrichFromIntrospectURL(ctx context.Context, s *sessions.SessionState) error {
+	clientSecret, err := p.GetClientSecret()
+	if err != nil {
+		return err
+	}
+	params := url.Values{}
+	params.Add("token", s.AccessToken)
+	basicAuth := b64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", p.ClientID, clientSecret)))
+	result := requests.New(p.IntrospectURL.String()).
+		WithContext(ctx).
+		WithMethod("POST").
+		WithBody(bytes.NewBufferString(params.Encode())).
+		SetHeader("Authorization", fmt.Sprintf("Basic %s", basicAuth)).
+		SetHeader("Content-Type", "application/x-www-form-urlencoded").
+		Do()
+
+	if result.StatusCode() != http.StatusOK {
+		return fmt.Errorf("error while requesting introspect claims, status code - %d", result.StatusCode())
+	}
+	s.IntrospectClaims = b64.StdEncoding.EncodeToString(result.Body())
```

Comment:
Here's my rationale for ditching the Base64 here: https://github.com/oauth2-proxy/oauth2-proxy/pull/1189#discussion_r810659434